### PR TITLE
Feat/draggable

### PR DESCRIPTION
**Pull request by tagomaru**
Hi @GNSPS 

This is a pull request for issue #42.
Could you check it and merge this?

### CODE REVIEW

**Comment by GNSPS** [comment]
The issue related to this PR is now funded @tagomaru  😉 

There's a few requests from my side:

Could you make it work similar to the other commands? This meaning that it would take a glob pattern/collection of files and work on them as opposed to the need of specifying the Truffle root.

Also I feel this would be much easier if you just implemented it as a flag/option on the `dependencies` command because basically what differs from that one is just the graph generation and you can just pick up after the `dependencies` array is populated.

Let me know what you think and if the funding is not representative of the workload.

**Comment by tagomaru** [comment]
Hi @GNSPS

Please let me make sure about what you mean.
Sample contracts are the below.

https://github.com/tagomaru/solidity-visualizer/tree/master/test/project/TGCrowdsale/contracts

Here, do you expect the command like below?

```
 1. $ surya inherianddepen contracts/*.sol
```

In this case, this command generates  the below html reports.

 * TGCrowdsale_timestamp.html
 * Migrations_timestamp.html
 * TGToken_timestamp.html

In the case of this directory structure, this work.

Howerver I am worried about the case that user custom contracts are deployed on directory like "contracts/**ICO**".

For example,
You can see the below import senetece in TGCrowdsale.sol

  import "openzeppelin-solidity/contracts/token/ERC20/MintableToken.sol";

Then, assume the below command.

```
2. $ surya inherianddepen ICO/*.sol
```

In the case of 1, tool understands that zeppelin directory is node_modules/openzeppelin-solidity from one upper directorie from target sol file.

// dir structrue
TGCrowdsale
│
├─contracts
│      Migrations.sol
│      TGCrowdsale.sol
│      TGToken.sol
│
├─node_modules
│  └─openzeppelin-solidity
│      │
│      ├─contracts

In the case of 2, tool makes wrong to decide zeppelin dir...

That is why I make users set project dir of truffle.

Exactlly, if I assume that users execute command only on project directory, this problem does not happen.

Please let me know your idea.

Thanks!

**Comment by GNSPS** [comment]
You can always use different glob patterns like:

```
$ surya inherianddepen contracts/**/*.sol
```

which traverses every folder inside `contracts/`.

For the `node_modules` the solution looks more like the work that @anurag-git did with this commit: https://github.com/ConsenSys/surya/pull/41/commits/892817a45a724cbb0868654d8bcc0ea56ca1d28f

**Comment by GNSPS** [comment]
Also I really feel that this could be done inside the current `graph` command.

**Comment by tagomaru** [comment]
Hi @GNSPS 

Thanks! I got it!

node_module solution looks great.
I am working on second PR now.
Please wait a few days.

**Comment by tagomaru** [comment]
Hi @GNSPS 

> Also I really feel that this could be done inside the current `graph` comma

Do you expect I should modify `graph.js` module instead of creating new module `inherianddepen.js`?

graph is for control-flow, so I expect creating new module.

**Comment by GNSPS** [comment]
I meant the `dependencies` one, sorry, @tagomaru! 😅

Just because they ultimately print the exact same thing

**Comment by tagomaru** [comment]
Hi @GNSPS 

[https://github.com/ConsenSys/surya/pull/44/commits/bfeaf92cd8971742b31cef32fa5956fe2f29ce07](url)

I handled the below on this PR.

- glob patterns

- automatically detect installed directory like 'zeppelin'

**Note**
This inherianddepen is for both of inheritance and using to understand those at a glance. That is why I feel it is better to separate this from existing modules such as graph, inheritance, and so on and added this as new feature.
And if we use graphviz, I guess it is difficult to process inheritance and using in same module...
If you still hope draggable inheritance and use graph are separatelly implemented in existing modules, please let me know.

And could you check and solve this conflict?

**Comment by tagomaru** [comment]
Hi @GNSPS

How about this PR?
If you have any concerns, pls let me know.
I can modify source code if you hope.

**Comment by GNSPS** [comment]
Ok, I think this is better. Still has some rough edges when the time to integrate into the whole Surya system comes. Especially since this graph is representing the same information as the `inheritance` command I think this can be a flag in that command.

This being said I really appreciate the help and think I can take it from here. So I'd invite you to point this PR to a new branch that I'm creating in the repo: `feat/draggable` so I can later accept the issue in Gitcoin. 😄

**Comment by tagomaru** [comment]
Hi @GNSPS 

> this graph is representing the same information as the inheritance

I do not think so because the new report shows not only inheritance relatioship but also both usinig declaration and userdefined type. And also this new report can consider package's contract automatically like zeppelin.

If you accept a new report which shows only inheritance relation, I will submit new PR for it since it is easy to expand the existing inheritance module with new option flag.

New PR module will load output of "surya inheritance hoge.sol" and generate html report with the output.
It means it is same as the part of output dot-formatted text file but only genrating report part is difference.

**Comment by tagomaru** [comment]
Hi @maurelian 

Thank you for your review.
I submitted new [PR](https://github.com/ConsenSys/surya/pull/44/commits/20be3642bbc33dc9dedbae509384bc6c9b9b6db6) for generating draggable report by inheritance command with option '-d'.

Do you mean deteciting package by improvement output?
If so, I will also improvement inheritance command.

File: package.json [hunk 1]
```diff
@@ -28,8 +28,6 @@
   "dependencies": {
     "c3-linearization": "^0.3.0",
     "colors": "^1.3.0",
-    "detect-installed": "^2.0.4",
```
Comment (maurelian) [inline package.json]: I had to reinstall detect installed.

**Comment by maurelian** [comment]
I think this is an improvement. 
@GNSPS what do you think about generating the output? Should it just print the html to standard out (console.log) or should there be an option to write to a file in a specified location?

**Comment by tagomaru** [comment]
@maurelian 
Thanks. I tetatively updated package.json for detect-installed.

File: package.json (hunk 1 above)
Comment (tagomaru) [reply package.json]: OK, I reinstalled.

**Comment by GNSPS** [comment]
Since the default behavior has always been printing to the console directly I don't think there's anything wrong with having that same behavior here. Maybe we should think about changing that globally.

This LGTM!

**Comment by maurelian** [comment]
see #67

**Pull request closed without merging**